### ISSUE TITLE
Windows Console now supports Sixel

### ISSUE THREAD

**Issue by NickAcPT** [smasher164/arewesixelyet#33]
It seems that Sixel support in the Windows Terminal is being worked on, and a PR for Sixel support in conhost has been merged (PR https://github.com/microsoft/terminal/pull/17421).

**Comment by j4james** [comment]
Just to be clear, that PR adds Sixel support to _conhost_ (that's the old Windows console). It's not available in Windows Terminal yet.

**Comment by NickAcPT** [comment]
Whoops, my bad! In that case however, does the Windows Console section need updating then?

EDIT: I went ahead and updated the title + initial post to reflect the actual changes I mentioned.

**Comment by j4james** [comment]
I'm not sure what exactly the criteria are for claiming sixel support on this site. Right now we've got a PR committed, and technically anyone could download the nightly build and use it if they really wanted to, but it'll be some time before it makes its way into the inbox Windows console which most users have access to. So for now you may want to hold off on claiming anything, or at least include a massive disclaimer clarifying that it's not generally available (and it's possible it might never be).

**Comment by j4james** [comment]
FYI, with PRs [microsoft/terminal/#17581](https://github.com/microsoft/terminal/pull/17581) and [microsoft/terminal/#17510](https://github.com/microsoft/terminal/pull/17510) having been merged now, we've got Sixel support in Windows Terminal too. As with conhost, it's still only accessible in the nightly build, but my understanding is that they are planning for a preview release around the end of the month.

**Comment by Kreijstal** [comment]
the nightly release is there, so we wait till stable releasse

**Comment by codr1** [comment]
I have it working in Windows Terminal Preview.

**Comment by TMA-2** [comment]
Yes, as of Windows Terminal Preview 1.22, [sixel support is now included](https://devblogs.microsoft.com/commandline/windows-terminal-preview-1-22-release/#sixel-image-support).

**Comment by AriESQ** [comment]
1.22 is still pre-release, the current release branch is 1.21.  

https://github.com/microsoft/terminal/releases

Searching through the Changelog I don't believe the sixel changes landed in the 1.21 branch.

**Comment by sitiom** [comment]
Now on stable as of https://github.com/microsoft/terminal/releases/tag/v1.22.10352.0

**Comment by CompeyDev** [comment]
Now what I'm a bit puzzled about is how on WSL querying capabilities seems to return no sixel support, however on native shells, it does return as supported. Odd...

**Comment by j4james** [comment]
@CompeyDev How are you querying the terminal capabilities? Sixel support can be detected with a [device attributes](https://vt100.net/docs/vt510-rm/DA1.html) report. To confirm that's working, you can execute the following in a WSL shell:
```
printf "\e[c"; read
```

And you should see something like this:
```
^[[?61;4;6;7;14;21;22;23;24;28;32;42c
```

The `4` in that response indicates that the terminal has support for Sixel.

**Comment by CompeyDev** [comment]
Yes, directly reading a sixel image from a file, or querying the capabilities tells me what sixel is supported and functional. However, if I were to use any programs which try to render images to the stdout using sixel, they all do nothing or fallback to other rendering methods.

**Comment by j4james** [comment]
@CompeyDev Can you give specific examples of apps that don't work for you? I've successfully used [`img2sixel`](https://github.com/libsixel/libsixel), [`timg`](https://github.com/hzeller/timg), and [`chafa`](https://github.com/hpjansson/chafa/) from within a WSL session. Although now that I think about, it's possible that older versions may have required a command line parameter to force Sixel output (e.g. `chafa -f sixel` or `timg -p sixel`). The latest versions should detect Sixel automatically, although you might need to build from source to get that functionality.

Another thing that might be relevant: I believe there was a bug in the new conpty system in version v1.22.10352 of Windows Terminal, which might have prevented query sequences from working. So if you haven't upgraded since then, I'd recommend trying a more recent version.

**Comment by CompeyDev** [comment]
I've tried `img2sixel` and `chafa` too -- but I could be on an older version which does not correctly infer it. Using `--force` flags do seem to work (as I think I previously mentioned as well; my comment may have been misunderstood). I'm on the latest version of WT too, so the issue likely is to do with the tools themselves not correctly inferring support.

**Comment by AriESQ** [comment]
Can issues be renamed? I believe this issue is for[ Microsoft Terminal](https://github.com/microsoft/terminal). The issue title refers to Microsoft Console, which is maybe confusing with the old `conhost.exe`.

**Comment by j4james** [comment]
@AriESQ It's kind of both. The issue was opened when Sixel support was added to OpenConsole, which is conhost. Later comments reported when support was added to Windows Terminal as well.

Not that it matters much either way. This repo appears to have been abandoned. Last update was more than a year ago.

**Comment by janantos** [comment]
created pull request for Windows Terminal https://github.com/smasher164/arewesixelyet/pull/53 as I have tested it in current install of win11 arm64 with default install of windows terminal and using wsl2 and cat sixel data directly from file

**Comment by miaLDN** [comment]
Sixel works perfectly in Windows Terminal (current release):

Version: 1.22.12111.0

``` sh
 $ printf "\e[c"; read
^[[?61;4;6;7;14;21;22;23;24;28;32;42;52c
```


https://github.com/microsoft/terminal/issues/448

**Comment by phuclv90** [comment]
Seems like this repo has been abandoned, no more commits since last year

**Comment by Kreijstal** [comment]
> Seems like this repo has been abandoned, no more commits since last year

true that

**Comment by miaLDN** [comment]
> Seems like this repo has been abandoned, no more commits since last year

:(

Gonna unfollow and unstar then :(

**Comment by smasher164** [comment]
Thanks for filing this! Windows Terminal is now marked as supported since 1.22, and the Windows Console entry explains that the built-in conhost.exe only has SIXEL in Insider Canary builds so far.
